contracts/utils/NoncesKeyed.sol

```diff
@@ -7,6 +7,10 @@ import {Nonces} from "./Nonces.sol";
  * @dev Alternative to {Nonces}, that supports key-ed nonces.
  *
  * Follows the https://eips.ethereum.org/EIPS/eip-4337#semi-abstracted-nonce-support[ERC-4337's semi-abstracted nonce system].
+ *
+ * NOTE: This contract inherits from {Nonces} and reuses its storage for the first nonce key (i.e. `0`). This
+ * makes upgrading from {Nonces} to {NoncesKeyed} safe. Doing so will NOT reset the current state of nonces to
+ * avoid replay attacks where a nonce is reused after the upgrade.
```

Comment:
```suggestion
 * makes upgrading from {Nonces} to {NoncesKeyed} safe. Doing so will NOT reset the current state of nonces.
```
This sentence is still a run on. Is it necessary to spell out the specific attack vector here?

Reply:
I don't think is necessary to mention an attack vector. Also, is there any in concrete?
Documenting it here is important for other systems (e.g. frontends or indexers) to know what behavior they should rely on.

Reply:
To be clear, I'm fine with the suggestion but perhaps @Amxx has a reason to keep the original

Reply:
This sentence wasn't affected.